core/domain/exp_jobs_one_off.py

```diff
@@ -35,8 +35,72 @@
 import python_utils
 import utils
 
-(exp_models,) = models.Registry.import_models([
-    models.NAMES.exploration])
+(
+    base_models,
+    exp_models,
+    feedback_models,
+    improvements_models,
+    skill_models,
+    stats_models,
+    story_models,
+) = models.Registry.import_models([
+    models.NAMES.base_model,
+    models.NAMES.exploration,
+    models.NAMES.feedback,
+    models.NAMES.improvements,
+    models.NAMES.skill,
+    models.NAMES.statistics,
+    models.NAMES.story,
+])
+
+
+class RegenerateStringPropertyIndexOneOffJob(
+        jobs.BaseMapReduceOneOffJobManager):
+    """One-off job for regenerating the index of models changed to use an
+    indexed StringProperty.
+
+    Cloud NDB dropped support for StringProperty(indexed=False) and
+    TextProperty(indexed=True). Therefore, to prepare for the migration to Cloud
+    NDB, we need to have every model that has been changed to regenerate their
```

Comment:
... we need to regenerate the indexes for every model that has been changed in this way.

Reply:
Done.